### PR TITLE
[XLA:CPU]  Moving oneDNN threadpool wrapper to tsl 

### PR DESCRIPTION
**Pull request by agramesh1**
This PR moves the oneDNN threadpool wrapper code under tsl, and uses tsl namesapce so it can also be called from XLA code.  
Changes include
1)	Removed dependency on reading an TF environment flag for using calling thread. The call  for reading the flag  (ThreadPoolUseCallerThread()) is removed and the env value is now passed into the constructor from TensorFlow.
2)	Pointer to the Eigen threadpool is directly passed to the constructor instead of getting it from OptKernelContext.
3)	Some additional changes for clarity and using oneDNN instead of mkl in variables and filenames.

### CODE REVIEW

File: tensorflow/core/kernels/mkl/mkl_avgpooling_op.cc [hunk 1]
```diff
@@ -123,7 +123,14 @@ class MklAvgPoolingOp : public MklPoolingForwardOpBase<T> {
...
           this->native_format_);
-      MklDnnThreadPool eigen_tp(context);
+      // Create the oneDNN wrapper over eigen threapool and set max threads
+      // in oneDNN.
+      Eigen::ThreadPoolInterface* eigen_interface =
+          context->device()
+              ->tensorflow_cpu_worker_threads()
+              ->workers->AsEigenThreadPool();
+      tsl::OneDnnThreadPool eigen_tp(eigen_interface,
+                                     ThreadPoolUseCallerThread());
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_avgpooling_op.cc:133]: Can we define an `EigenThreadPoolFromTfContext` macro or inline function in `mkl_util.h` (or somewhere convenient)? Then we can create the threadpool in one statement.
```suggestion
      tsl::OneDnnThreadPool eigen_tp(EigenThreadPoolFromTfContext(context),
                                     ThreadPoolUseCallerThread());
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you very much for the PR, @agramesh1! The overall PR looks good to me. I have two minor comments.

**Comment by agramesh1** [comment]
> Thank you very much for the PR, @agramesh1! The overall PR looks good to me. I have two minor comments.

@penpornk I made the changes. Please take a look at it.

**Review by penpornk — APPROVED** [review]
Thank you for the changes!

**Pull request merged**